File: app/src/main/java/com/subhrajyoti/wallify/background/SetWallpaperTask.java
package com.subhrajyoti.wallify.background;

import android.graphics.Bitmap;
import android.os.AsyncTask;

import com.subhrajyoti.wallify.Utils;

import java.io.IOException;

public class SetWallpaperTask extends AsyncTask<Bitmap, Void, Boolean> {

    @Override
    protected Boolean doInBackground(Bitmap... params) {
        boolean status= true;
        final Bitmap bitmap = params[0];
        try {
            Utils.getWallpaperManager().setBitmap(bitmap);
        } catch (IOException e) {
            status = false;
            e.printStackTrace();
        }
        return status;
    }
}

File: app/src/main/java/com/subhrajyoti/wallify/background/SetWallpaperTask.kt
package com.subhrajyoti.wallify.background

import android.graphics.Bitmap
import android.os.AsyncTask
import com.subhrajyoti.wallify.Utils.wallpaperManager
import java.io.IOException

class SetWallpaperTask : AsyncTask<Bitmap?, Void?, Boolean>() {
    override fun doInBackground(vararg params: Bitmap?): Boolean? {
        var status = true
        val bitmap = params[0]
        try {
            wallpaperManager.setBitmap(bitmap)
        } catch (e: IOException) {
            status = false
            e.printStackTrace()
        }
        return status
    }
}